internal/stores/metadata.go

```diff
@@ -481,30 +489,51 @@ func (s *SQLStore) SetContractSet(ctx context.Context, name string, contractIds
 	return s.db.Model(&contractset).Association("Contracts").Replace(&dbContracts)
 }
 
-func (s *SQLStore) DeleteContractSet(ctx context.Context, name string) error {
-	return s.db.
-		Where(dbContractSet{Name: name}).
-		Delete(&dbContractSet{}).
-		Error
-}
-
 func (s *SQLStore) RemoveContract(ctx context.Context, id types.FileContractID) error {
-	err := removeContract(s.db, fileContractID(id))
+	contract, err := s.contract(ctx, fileContractID(id))
 	if err != nil {
 		return err
 	}
+
+	if err := s.retryTransaction(func(tx *gorm.DB) error {
+		return archiveContracts(tx, []dbContract{contract}, api.ContractArchivalReasonRemoved)
+	}); err != nil {
+		return err
+	}
+
 	delete(s.knownContracts, id)
 	return nil
 }
 
 func (s *SQLStore) RemoveContracts(ctx context.Context) error {
```

Comment:
I'd rename this method to `ArchiveAllContracts` since having both `ArchiveContracts` and `RemoveContracts` is a bit confusing since they both archive.